### ISSUE TITLE
Updated analysis: rerun GISTIC with consensus SEG file

### ISSUE THREAD

**Issue by jaclyn-taroni** [AlexsLemonade/OpenPBTA-analysis#453]
#### What analysis module should be updated and why?

_Note: Running GISTIC is upstream of this repository._

We now have consensus copy number calls (#128). Furthermore, we have a consensus SEG file that contains the required information to run GISTIC (discussed in #392 and implemented in #441): `analyses/copy_number_consensus_call/results/pbta-cnv-consensus.seg`  

#### What changes need to be made? Please provide enough detail for another participant to make the update.

I believe this is as straightforward as switching what `segfile` points to here: https://github.com/d3b-center/publication_workflows/blob/master/openPBTA/run-gistic.sh

#### What input data should be used? Which data were used in the version being updated?

Input: `analyses/copy_number_consensus_call/results/pbta-cnv-consensus.seg` 

In the GISTIC output that's included in the data download, the `pbta-cnv-cnvkit.seg` file was used as input. That file has not changed in some time according to the [release notes](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/doc/release-notes.md).

#### When do you expect the revised analysis will be completed?

Unknown.

#### Who will complete the updated analysis?

Tagging @jharenza as this is upstream of the repo. The GISTIC output may be included in the v14 release (#432), but ideally this can be made available ahead of time so it can start getting used in downstream analyses (see: #8).

**Comment by jharenza** [comment]
Hi @jaclyn-taroni! I set this up, but it quickly failed. 
`[harenzaj@c-2-02 PBTA]$ bash run_gistic_consensus.sh 
--- running GISTIC ---
GISTIC version 2.0.23
required parameters:
      base_dir: /mnt/isilon/maris_lab/target_nbl_ngs/PBTA/gistic/2019-01-18-consensus-cnv/
       segfile: /mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg
marker spacing: 10000
   refgenefile: /mnt/isilon/maris_lab/target_nbl_ngs/Gonzalo/install/GISTIC_2_0_23/refgenefiles/hg38.UCSC.add_miR.160920.refgene.mat
optional_params = 
         array_list_file: []
                cnv_file: []
                   t_amp: 0.1000
                   t_del: 0.1000
       join_segment_size: 2
                     ext: []
               qv_thresh: 0.2500
                remove_X: 0
                 markers: []
      max_marker_spacing: 10000
      run_broad_analysis: 1
        broad_len_cutoff: 0.9800
                   ziggs: [1x1 struct]
                     res: 0.0500
              conf_level: 0.9000
                     cap: 1.5000
          do_gene_gistic: 1
     conserve_disk_space: 0
         save_data_files: 1
            use_segarray: 1
        write_gene_files: 1
           use_two_sided: 1
          do_arbitration: 1
           save_seg_data: 1
                   fname: []
              peak_types: {'robust'}
             genepattern: 1
             arm_peeloff: 1
    gene_collapse_method: 'extreme'
           sample_center: 'median'

Checking inputs...
Making D struct
No markers file specified: generating pseudo-markers!
Reading Seg File '/mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg'
 
GISTIC 2.0 input error detected:
39 segment overlaps detected in file '/mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg'.
First overlap detected between segments at lines 3662 and 176.`

GISTIC does not like overlapping segments - I am not sure if this is something that @jashapiro can address. It is annoying, and I think we actually had this issue before and would recreate segments such that they would not overlap, but seems like it may be a better thing to do through GitHub with review.

**Comment by jashapiro** [comment]
I will look into this next week.

**Comment by jashapiro** [comment]
Okay, I couldn't resist, and started to look at it now. It appears that this is the result of overzealous merging of nearby CNVs, which results in trouble when there are multiple CNVs in opposite directions nearby. I will look into tweaking the consensus calls to catch and separate these, as well as other causes that might be in the mix.

**Comment by jaclyn-taroni** [comment]
Now that #463 is merged, `analyses/copy_number_consensus_call/results/pbta-cnv-consensus.seg` is ready to be run through GISTIC.

**Comment by jharenza** [comment]
on it!

**Comment by jharenza** [comment]
Ahh, @jashapiro - while the overlaps are no longer giving errors, it looks like the data are incomplete, in that GISTIC expects that all regions are the same across all samples, including regions with neutral CN. See [this post](https://www.biostars.org/p/254455/). I think this fix could be as simple as adding all missing regions back as CN 2.

```
[harenzaj@c-2-02 PBTA]$ bash run_gistic_consensus.sh
mkdir: cannot create directory ‘/mnt/isilon/maris_lab/target_nbl_ngs/PBTA/gistic/2019-01-22-consensus-cnv/’: File exists
--- running GISTIC ---

GISTIC version 2.0.23
required parameters:
      base_dir: /mnt/isilon/maris_lab/target_nbl_ngs/PBTA/gistic/2019-01-22-consensus-cnv/
       segfile: /mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg
marker spacing: 10000
   refgenefile: /mnt/isilon/maris_lab/target_nbl_ngs/Gonzalo/install/GISTIC_2_0_23/refgenefiles/hg38.UCSC.add_miR.160920.refgene.mat
optional_params = 
         array_list_file: []
                cnv_file: []
                   t_amp: 0.1000
                   t_del: 0.1000
       join_segment_size: 2
                     ext: []
               qv_thresh: 0.2500
                remove_X: 0
                 markers: []
      max_marker_spacing: 10000
      run_broad_analysis: 1
        broad_len_cutoff: 0.9800
                   ziggs: [1x1 struct]
                     res: 0.0500
              conf_level: 0.9000
                     cap: 1.5000
          do_gene_gistic: 1
     conserve_disk_space: 0
         save_data_files: 1
            use_segarray: 1
        write_gene_files: 1
           use_two_sided: 1
          do_arbitration: 1
           save_seg_data: 1
                   fname: []
              peak_types: {'robust'}
             genepattern: 1
             arm_peeloff: 1
    gene_collapse_method: 'extreme'
           sample_center: 'median'

Checking inputs...
Making D struct
No markers file specified: generating pseudo-markers!
Reading Seg File '/mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg'
Warning: Shortened 284 segments in '/mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg' that overlap by one marker.
> In make_D_from_segseq_data at 122
  In run_gistic2_from_seg at 193
  In gp_gistic2_from_seg at 97 
D = 
               sdesc: {713x1 cell}
                chrn: [309494x1 double]
                 pos: [309494x1 double]
    suppress_history: 1
               islog: 0
                isMB: 0
                 dat: [309494x713 SegArray]

D = 
               sdesc: {713x1 cell}
                chrn: [309494x1 double]
                 pos: [309494x1 double]
    suppress_history: 1
               islog: 0
                isMB: 0
                 dat: [309494x713 SegArray]

Matrix size 309494     713
Removing NaN probes...
Removing 309494 markers with NaNs
Matrix size 0  713
 
GISTIC 2.0 input error detected:
All input data were removed after NaN processing.
```

**Comment by jashapiro** [comment]
Okay, I could do that. But I kind of don't want to, unless we have some way to capture the fact that there are uncalled regions in there (i.e. regions where we have no confidence in any particular call). Can I set copy number to NA at all, or will gistic choke on that as well? Also, note that some of the regions are already NA for the seg.mean field. Does that cause trouble for gistic?

**Comment by jharenza** [comment]
Good point - actually when we normally would export these regions from snp array segmentation software, we would have to trick it so all segments without a CN change would be neutral - we did not end up using NAs - let me try adding some NAs on an old seg file we know works - standby.

**Comment by jharenza** [comment]
@jashapiro - yes, that will work - tested on the old cnvkit seg file, and run finishes successfully!

**Comment by jharenza** [comment]
Using [this](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/copy_number_consensus_call/results/pbta-cnv-consensus.seg) seg file, we still have overlapping segments.

```
[harenzaj@c-2-01 PBTA]$ bash run_gistic_consensus.sh
--- running GISTIC ---
GISTIC version 2.0.23
required parameters:
      base_dir: /mnt/isilon/maris_lab/target_nbl_ngs/PBTA/gistic/2019-01-28-consensus-cnv/
       segfile: /mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg
marker spacing: 10000
   refgenefile: /mnt/isilon/maris_lab/target_nbl_ngs/Gonzalo/install/GISTIC_2_0_23/refgenefiles/hg38.UCSC.add_miR.160920.refgene.mat
optional_params = 
         array_list_file: []
                cnv_file: []
                   t_amp: 0.1000
                   t_del: 0.1000
       join_segment_size: 2
                     ext: []
               qv_thresh: 0.2500
                remove_X: 0
                 markers: []
      max_marker_spacing: 10000
      run_broad_analysis: 1
        broad_len_cutoff: 0.9800
                   ziggs: [1x1 struct]
                     res: 0.0500
              conf_level: 0.9000
                     cap: 1.5000
          do_gene_gistic: 1
     conserve_disk_space: 0
         save_data_files: 1
            use_segarray: 1
        write_gene_files: 1
           use_two_sided: 1
          do_arbitration: 1
           save_seg_data: 1
                   fname: []
              peak_types: {'robust'}
             genepattern: 1
             arm_peeloff: 1
    gene_collapse_method: 'extreme'
           sample_center: 'median'
Checking inputs...
Making D struct
No markers file specified: generating pseudo-markers!
Reading Seg File '/mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg'
GISTIC 2.0 input error detected:
294 segment overlaps detected in file '/mnt/isilon/maris_lab/target_nbl_ngs/PBTA/pbta-cnv-consensus.seg'.
First overlap detected between segments at lines 1358 and 1391.
```

**Comment by jashapiro** [comment]
I was wrong in thinking the version in master was correct at this point. I had actually not updated the files, as at the time I didn't want a v14 based file in the master branch. The current version is attached here. If it works, I will update the master branch.

[pbta-cnv-consensus.seg.gz](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4123124/pbta-cnv-consensus.seg.gz)

**Comment by jharenza** [comment]
This file ran successfully. GISTIC results below, and will include in #432 

[2019-01-28-consensus-cnv.zip](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4123481/2019-01-28-consensus-cnv.zip)

**Comment by jharenza** [comment]
closed via #507
